src/sentry/integrations/slack/sdk_client.py

```diff
@@ -35,16 +39,43 @@ def track_response_data(response: SlackResponse, method: str, error: str | None
     logger.info("integration.http_response", extra=extra)
 
 
-def wrapper(method: FunctionType):
+def is_response_fatal(response: SlackResponse) -> bool:
+    if not response.get("ok"):
+        if "account_inactive" == response.get("error", ""):
+            return True
+    return False
+
+
+def record_response_for_disabling_integration(response: SlackResponse, integration_id: int) -> None:
+    redis_key = f"sentry-integration-error:{integration_id}"
+
+    buffer = IntegrationRequestBuffer(redis_key)
+    if is_response_fatal(response):
+        buffer.record_fatal()
+    else:
+        if is_response_success(response):
+            buffer.record_success()
+            return
+        if is_response_error(response):
+            buffer.record_error()
+    if buffer.is_integration_broken():
+        disable_integration(buffer, redis_key, integration_id)
+
+
+def wrapper(method: FunctionType, integration_id):
     @wraps(method)
     def wrapped(*args, **kwargs):
+        if method.__name__ not in SLACK_SDK_WRAP_METHODS:
+            return method(*args, **kwargs)
+
         try:
             response = method(*args, **kwargs)
-            if isinstance(response, SlackResponse):
-                track_response_data(response=response, method=method.__name__)
+            track_response_data(response=response, method=method.__name__)
         except SlackApiError as e:
             if e.response:
                 track_response_data(response=e.response, error=str(e), method=method.__name__)
+
+                record_response_for_disabling_integration(e.response, integration_id)
```

Comment:
@ceorourke does this need to record all responses or just error responses? i notice in the existing logic we seem to only be recording success responses. the `record_response_for_disabling_integration` function isn't in any of the except statements https://github.com/getsentry/sentry/blob/e796c94b46e5fa06bf75eb21f201619726aba07b/src/sentry/shared_integrations/client/base.py#L298

Reply:
I think it should just be error responses, I see what you're saying but I dont think it'd be working at all if it's only recording successes 🤔 and we know it is disabling stuff from checking logs

Reply:
okay the spec here says it should be error and successes https://www.notion.so/sentry/Tech-Spec-Notify-on-Disabled-Integration-Spec-e7ea0f86ccd6419cb3e564067cf4a2ef